### PR TITLE
Irate delta fix

### PR DESCRIPTION
**Pull request by srivik**
**Pull Request checklist**

- [ ] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [ ] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :**
When there is only one sample available for delta counters during irate function, we do not return any result for the same since we do not have info about the publish interval. 

**New behavior :**
When there is only one sample for delta counters, we will assume a default 60 second publish interval to return the value for that instant.

### CODE REVIEW

**Review by alextheimer — DISMISSED** [review]
LGTM 👍 -- Couple questions/comments, but none are blockers.

File: query/src/main/scala/filodb/query/exec/rangefn/RangeInstantFunctions.scala [hunk 1]
```diff
@@ -139,17 +139,20 @@ object IRatePeriodicFunction extends RangeFunction {
             window: Window,
             sampleToEmit: TransientRow,
             queryConfig: QueryConfig): Unit = {
-    if (window.size < 2) {
-      sampleToEmit.setValues(endTimestamp, Double.NaN) // cannot calculate result without 2 samples
-    } else {
-      lastFunc.apply(startTimestamp, endTimestamp, window, sampleToEmit, queryConfig)
+    if (window.size < 1) {
```
[the remark this answers is not in the document]
Comment (srivik) [reply query/src/main/scala/filodb/query/exec/rangefn/RangeInstantFunctions.scala:142]: Good point, I will ensure this gets tested for irate(gauge{})

File: query/src/main/scala/filodb/query/exec/rangefn/RangeInstantFunctions.scala [hunk 2]
```diff
@@ -139,17 +139,20 @@ object IRatePeriodicFunction extends RangeFunction {
...
+      sampleToEmit.setValues(endTimestamp, Double.NaN) // cannot calculate result without at least 1 sample
```
[the remark this answers is not in the document]
Comment (srivik) [reply query/src/main/scala/filodb/query/exec/rangefn/RangeInstantFunctions.scala:143]: Yeah good point

File: query/src/main/scala/filodb/query/exec/rangefn/RangeInstantFunctions.scala [hunk 3]
```diff
@@ -139,17 +139,20 @@ object IRatePeriodicFunction extends RangeFunction {
...
+    }
+    //If there is only one sample, default the sample interval to 60 seconds
+    var sampledInterval: Double = 60
```
[the remark this answers is not in the document]
Comment (srivik) [reply query/src/main/scala/filodb/query/exec/rangefn/RangeInstantFunctions.scala:146]: Sher and I discussed this option offline and decided to not make it configurable.

**Review by alextheimer — APPROVED** [review]
Many thanks for the updates 🙏 -- LGTM

**Pull request merged**